### ISSUE TITLE
Top CPU and RAM consumers

### ISSUE THREAD

**Issue by asm0dey** [henrygd/beszel#44]
Currently, I can see that CPU usage grew, but do not have a way to understand who's the consumer. It would be nice to add this information as well!

**Comment by henrygd** [comment]
Yeah, it's apparent when the culprit is a docker container, but not so much if it's a regular process.

I want to keep the amount of data exchanged and stored to a minimum, so my opinion is that sending info on every process is too much.

But I can look into possibly logging the top five cpu processes.

**Comment by asm0dey** [comment]
Probably it might be a tunable parameter. For me even top 3 will work, I can imagine a situation when somebody needs 10

**Comment by Buco7854** [comment]
+1 for this. Would love to be able to see process the same way we already do with container. As mentioned i think it would be more user friendly to allow for any number (defined by the user with an env) of process based on the CPU if in CPU and RAM depending on the graph.

**Comment by bl4ckswordsman** [comment]
+1. Was about to post this same feature request but found this

**Comment by AndreiTelteu** [comment]
Instead of a dedicated process list maybe the same graph type that is used for docker cpu & memory usage would be perfect.
It's very easy to see at a glance what docker container is using my cpu at a quick hover.
It's simple and easy to use.

![Image](https://github.com/user-attachments/assets/e93e3a51-41b6-45bd-97a9-26587a7e0280)

![Image](https://github.com/user-attachments/assets/277c14ab-e73e-404e-8dc1-678b27c83543)

**Comment by svenvg93** [comment]
@henrygd Technically this is now possible with the systemd service monitoring right?

**Comment by henrygd** [comment]
I think the scope of this one is still a little broader, where we would want to collect processes that aren't tied to services.

**Comment by svenvg93** [comment]
Thanks for taking the time to suggest this!

This issue is a duplicate/similar of an existing one, so we’ll be closing it to keep the discussion in a single place. You can follow progress and updates on the linked issue, and feel free to subscribe there or add any additional details that might help.

Thanks again for helping improve the project!

**Comment by asm0dey** [comment]
Hah, my issue is #44, that one is #971, I guess that one is a duplicate of this one. But whatever works!

**Comment by svenvg93** [comment]
Technically, yes. Since #971 has the most recent activity, I opted to keep that one open. In the end, both will accomplish the same thing.
